### PR TITLE
Add rest request gzip decompress

### PR DESCRIPTION
**Pull request by gawsoftpl**
I add gzip decompress for requests if has header Content-Encoding

### CODE REVIEW

**Comment by gawsoftpl** [comment]
Fastapi handles GZipMiddleware but only for Response. For a request with gzip payload and header Content-encoding: gzip Fastapi requires additional Middleware to decompress that payload. 

https://fastapi.tiangolo.com/advanced/custom-request-and-route/#create-a-custom-gziprequest-class

File: mlserver/rest/gzip_request.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+
+class GzipRequest(Request):
+    async def body(self) -> bytes:
+        if not hasattr(self, "_body"):
+            body = await super().body()
+            if "gzip" in self.headers.getlist("Content-Encoding"):
+                body = gzip.decompress(body)
+            self._body = body
+        return self._body
```
Comment (adriangonz) [inline mlserver/rest/gzip_request.py:14]: We already have a custom `Request` object defined in https://github.com/SeldonIO/MLServer/blob/master/mlserver/rest/requests.py. Would it be possible to add the `body` method there, and remove the need for the `GzipRoute` router?

**Review by adriangonz — CHANGES_REQUESTED** [review]
Got it! Thanks for clarifying the difference.

I've added a small comment to simplify the implementation. Other than that, this looks good!

File: mlserver/rest/gzip_request.py (hunk 1 above)
Comment (gawsoftpl) [reply mlserver/rest/gzip_request.py:14]: I will check that, if that will work I will make a new pull request.

Comment (gawsoftpl) [reply mlserver/rest/gzip_request.py]: I create gzip decompress in Custom Request class (https://github.com/SeldonIO/MLServer/blob/master/mlserver/rest/requests.py.)
I will make new pull requests. I close this one

**Pull request closed without merging**